### PR TITLE
Bitcoind v0.21.0 support

### LINKED ISSUE #861: Update BIP 157 core branch to latest version
https://github.com/bitcoin/bitcoin/pull/16442 was rebased today and the update should be reflected here.

### LINKED ISSUE #1629: Update neutrino binary with latest bitcoin core code
Our neutrino binary is out of date, we should update to the latest functionality and test it

https://github.com/bitcoin/bitcoin/pull/18876

### PR DESCRIPTION
**Pull request by benthecarman**
Closes #2126 
Closes #1629
Closes #861

https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-0.21.0.md

### CODE REVIEW

File: bitcoind-rpc-test/src/test/scala/org/bitcoins/rpc/v21/BitcoindV21RpcClientTest.scala [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+import scala.concurrent.Future
+
+class BitcoindV21RpcClientTest extends BitcoindRpcTest {
+
+  lazy val clientPairF: Future[(BitcoindV21RpcClient, BitcoindV21RpcClient)] =
+    BitcoindRpcTestUtil.createNodePairV21(clientAccum)
+
+  lazy val clientF: Future[BitcoindV21RpcClient] = clientPairF.map(_._1)
+
+  clientF.foreach(c => clientAccum.+=(c))
```
Comment (Christewart) [inline bitcoind-rpc-test/src/test/scala/org/bitcoins/rpc/v21/BitcoindV21RpcClientTest.scala:25]: don't you want to add the `clientPairF` to the accumulator? Seems you are only adding one client here

File: testkit/src/main/scala/org/bitcoins/testkit/rpc/BitcoindRpcTestUtil.scala [hunk 2]
```diff
@@ -103,6 +105,7 @@ trait BitcoindRpcTestUtil extends BitcoinSLogger {
       if (blockFilterIndex)
         conf + """
                  |blockfilterindex=1
+                 |peerblockfilters=1
```
Comment (Christewart) [inline testkit/src/main/scala/org/bitcoins/testkit/rpc/BitcoindRpcTestUtil.scala:108]: what does this setting do?

**Review by Christewart — CHANGES_REQUESTED** [review]
I vote we don't merge until the official v21 is released

File: testkit/src/main/scala/org/bitcoins/testkit/rpc/BitcoindRpcTestUtil.scala (hunk 2 above)
Comment (benthecarman) [reply testkit/src/main/scala/org/bitcoins/testkit/rpc/BitcoindRpcTestUtil.scala:108]: This makes it so we can send block filters over the network

File: bitcoind-rpc-test/src/test/scala/org/bitcoins/rpc/v21/BitcoindV21RpcClientTest.scala (hunk 1 above)
Comment (benthecarman) [reply bitcoind-rpc-test/src/test/scala/org/bitcoins/rpc/v21/BitcoindV21RpcClientTest.scala:25]: `createNodePairV21` adds  them to it

File: bitcoind-rpc/bitcoind-rpc.sbt [hunk 3]
```diff
@@ -51,7 +51,10 @@ TaskKeys.downloadBitcoind := {
     val location =
       if (version == experimentalVersion)
         s"https://s3-us-west-1.amazonaws.com/suredbits.com/bitcoin-core-$version/bitcoin-$version-$platform.$suffix"
-      else
+      else if (version.init.endsWith("rc")) { // if it is a release candidate
```
Comment (Christewart) [inline bitcoind-rpc/bitcoind-rpc.sbt:54]: i think this can be removed now?

**Review by Christewart — APPROVED** [review]
This LGTM overall, i think there is just one thing that needs to be removed in the build, and then we should also remove the `@DoNotDiscover` annotation from this test case

https://github.com/bitcoin-s/bitcoin-s/blob/da54d2e9fbd06f5a9e569bfb89321cc4be97d5ba/node-test/src/test/scala/org/bitcoins/node/NeutrinoNodeTest.scala#L22

File: bitcoind-rpc/bitcoind-rpc.sbt (hunk 3 above)
Comment (benthecarman) [reply bitcoind-rpc/bitcoind-rpc.sbt:54]: imo we should leave this, might want to use a release canidate in the future and will just have to put this back

**Comment by Christewart** [comment]
Looks like we might be having some problems starting the node in some cases? 

https://github.com/bitcoin-s/bitcoin-s/pull/2414/checks?check_run_id=1703821908#step:5:988

**Pull request merged**